horace_core/sqw/coord_transform/@ubmat_proj/ubmat_proj.m

```diff
@@ -0,0 +1,387 @@
+    %                   :
+    %   >> proj = ubmat_proj(...,'lab4',labelstr,...)
+    %
+    % Minimal fully functional form:
+    %   >> proj =  ubmat_proj(u_to_rlu,'alatt',latice_parameters,'angdeg',lattice_angles_in_degrees);
+    %
+    %IMPORTANT:
+    % if you want to use ubmat_proj as input for the cut algorithm, it needs
+    % at least one input parameter u_to_rlu, (or its default value) as
+    % the lattice parameters for cut will be taken from sqw object
+    % if not provided with projection.
+    %
+    % For independent usage u_to_rlu and lattice parameters (minimal fully
+    % functional form) needs to be specified. Any other parameters have
+    % their reasonable defaults and need to change only if change in their default values
+    % is required.
+    %
+    % Input:
+    % ------
+    % Projection axes are defined by two vectors in reciprocal space, together
+    % with optional arguments that control normalisation, orthogonality, labels etc.
+    % The input can be a data structure with field-names and contents chosen from
+    % the arguments below, or alternatively the arguments
+    %
+    % Required arguments:
+    %   u_to_rlu
+    %
+    %
+    % Also accepts these and aProjectionBase properties as set of key-values
+    % pairs following standard serializable class constructor agreements.
+    %
+    % NOTE:
+    % constructor does not accept legacy ub_inv_legacy matrix, even if it is specified
+    % in the list of saveable properties.
+    %
+    properties(Dependent)
+        %-----------------------------------------------------------------
+        % DERIVED PROPERTIES:
+        u; %[1x3] Vector of first axis (r.l.u.)
+        v; %[1x3] Vector of second axis (r.l.u.)
+        w; %[1x3] Vector of third axis (r.l.u.) - used only if third character of type is 'p'
+        type;  % Character string which defines normalization and left
+        % for compartibility with line_proj
+        nonorthogonal; % Indicates if non-orthogonal axes are used (if true)
+        %
+        %
+        uoffset  % offset expressed in image coordinate system. Old interface to img_offset
+        % which is under construction. Transient property used to process
+        % input parameters which converted to offset and nullified after
+        % that.
+    end
+    properties(Dependent,Hidden)
+        % return set of vectors, which define primary lattice cell if
+        % coordinate transformation is non-orthogonal
+        unit_cell;
+    end
+    properties(Hidden)
+        % Developers option. Use old (v3 and below) sub-algorithm in
+        % ortho-ortho transformation to identify cells which may contribute
+        % to a cut. Correct value is chosen on basis of performance analysis
+        convert_targ_to_source=true;
+    end
+
+    properties(Access=protected)
+        % Cached properties value, calculated from input u_to_rlu matrix
+        type_  = 'aaa';
+        nonorthogonal_ = false;
+        uvw_cache_     = eye(4);
```

Comment:
Why does `uvw_cache` have 4 4-vectors. It can't just be `uvw` in that case.

Reply:
because it is how `u_to_rlu` was written in old code